syntax = "proto3";


service DialogService{
    rpc SendReply (Reply) returns (Reply);
    rpc GetReplyHistoryLimited (GetReplyHistoryLimitedParam) returns (ReplyHistoryLimited);
    rpc GetReplyHistory (GetReplyHistoryParam) returns (stream Reply);
}

message Reply{
    string speaker_id = 1;
    string comment = 2;
}

message GetReplyHistoryLimitedParam{
    string speaker_id = 1;
    int32 history_from = 2;
}

message GetReplyHistoryParam{
    string speaker_id = 1;
}

message ReplyHistoryLimited{
    repeated Reply replies = 3;

}

message Empty {
    int32 status_code = 1;
}